glusterd2/events/eventhandler.go

```diff
@@ -154,3 +162,69 @@ func (h *handler) Handle(e *api.Event) {
 func (h *handler) Events() []string {
 	return h.events
 }
+
+func getJWTToken(eventtype, secret string) string {
+	claims := &jwt.StandardClaims{
+		IssuedAt:  time.Now().Unix(),
+		ExpiresAt: time.Now().Add(time.Second * 10).Unix(),
+		Subject:   eventtype,
+	}
+
+	token := jwt.NewWithClaims(jwt.SigningMethodHS256, claims)
+	ss, err := token.SignedString([]byte(secret))
+	if err != nil {
+		return ""
+	}
+	return ss
+}
+
+//WebhookPublish sends event to registered webhooks
+func WebhookPublish(webhook *eventsapi.Webhook, e *api.Event) error {
+	message, err := json.Marshal(e)
+	if err != nil {
+		log.WithFields(log.Fields{
+			"name":  e.Name,
+			"error": err.Error(),
+		}).Error("failed to marshal event")
+		return err
+	}
+
+	body := strings.NewReader(string(message))
+
+	req, err := http.NewRequest("POST", webhook.URL, body)
+	if err != nil {
+		log.WithError(err).Error("error forming the request object")
+		return err
+	}
+	req.Header.Set("Content-Type", "application/json")
```

Comment:
It seems a bit odd to me that this api is assuming the other side uses json. Perhaps we should consider allowing per api serialization so that the caller can provide this with whatever body the api on the other side actually expects (perhaps the remote api works with xml for example)

Reply:
JSON is the only supported format now.